### ISSUE TITLE
[a11y] Review and improve a11y of save login alert

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/firefox-ios#24665]
Let’s review the a11y of the `SaveLoginAlert`. This has some investigation and might need some discussion (please comment in here with findings).

1. Should we have button layout change on bigger font sizes?
    - We should test with "longer" languages like German if the button sizes properly. If they don't we might need to change the layout on bigger `contentSizeCategory` (received with a `UIContentSizeCategory.didChangeNotification`).
1. Check `VoiceOver` is functioning as it should. 
    - We should be able to navigate this prompt with Voice Over (i.e. buttons are read as buttons and swiping on screen with VoiceOver activated navigates the prompt items one by one).

!image-20250210-184252.png|width=1179,height=2556,alt="image-20250210-184252.png"!

![Image](https://github.com/user-attachments/assets/5530dc3c-ec4a-4842-bd4b-1e0716073306)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-11334)

**Comment by anna-yeddi** [comment]
![image](https://github.com/user-attachments/assets/a478739a-fb0d-464d-bb7c-caeb8fe9adde)

It seems that in the current release the popup is not announced by VoiceOver, but it is the next focusable element at the end of the web content:
1. It would be much harder to uncover when a login page has a lot of content after the "Submit" button (or its alternative). On the screenshot, the element in visible VO focus is under the popup and it was the last activated button.
2. There is no indication where the save password question is coming from: website or Firefox, which may be confusing for users.

I'm marking this issue as `access-s3` but would highly recommend to resolve at least the item 1 (lack of focus management or lack of alert treatment).

Button layout does not have to be changed, from technical WCAG compliance perspective, while all button labels are visible and readable. But I agree this would be a nice experience improvement for users with large font sizes.
